beaker/src/shared_metadata.rs

```diff
@@ -7,6 +7,67 @@ use std::fs;
 use std::path::Path;
 use std::time::Instant;
 
+/// ONNX download cache statistics
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct OnnxCacheStats {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub model_cache_hit: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub download_time_ms: Option<f64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cached_models_count: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cached_models_size_mb: Option<f64>,
+}
+
+/// CoreML cache statistics
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct CoremlCacheStats {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cache_hit: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cache_count: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cache_size_mb: Option<f64>,
+}
+
+/// Cache statistics collected from cache operations (instrumentation only)
+#[derive(Debug, Default, Clone)]
```

Comment:
I don't think this struct or its impl should exist- delete both. Pass around the two objects separately.